test/jdk/java/foreign/TestLayouts.java

```diff
@@ -177,7 +186,17 @@ public void testAlignmentString(MemoryLayout layout, long bitAlign) {
         for (long a : alignments) {
             if (layout.bitAlignment() == layout.bitSize()) {
                 assertFalse(layout.toString().contains("%"));
-                assertEquals(layout.withBitAlignment(a).toString().contains("%"), a != bitAlign);
+                /* Ignored the check on '%' in the string representation of JAVA_DOUBLE on AIX
+                 * given '%' is skipped deliberately in decorateLayoutString() for JAVA_DOUBLE
+                 * on AIX when the adjusted alignment is 32, 64 or 128.
+                 *
+                 * See hasNaturalAlignment() in ValueLayouts.java and decorateLayoutString()
+                 * in AbstractLayout.java for more details.
+                 */
+                MemoryLayout alignedLayout = layout.withBitAlignment(a);
+                boolean isCheckRequired = (isAixOS && (layout == JAVA_DOUBLE)) ?
+                                                ((alignedLayout.bitAlignment() % 32) != 0) : (a != bitAlign);
```

Comment:
When would `alignedLayout.bitAlignment() != a`? I think it's guaranteed in which case this can simplified to
```java
                boolean naturalAlignment = (isAixOS && (layout == JAVA_DOUBLE)) ?
                                                ((a % 32) == 0) : (a == bitAlign);
                assertEquals(layout.withBitAlignment(a).toString().contains("%"), !naturalAlignment);
```
I think the comment above is misleading: we're not ignoring the check, we're adjusting our expectations based on special treatment of `double` on AIX. Perhaps
> Adjust the check for '%' in the string representation of JAVA_DOUBLE on AIX

Reply:
I agree to update against the code/comment above but the idea is literally the same given we don't check `%` in `JAVA_DOUBLE` string on AIX due to the alignment adjustment in code.